File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> combo = new ArrayList<>();

        combinationSum(nums, target, 0, res, combo);
        
        return res;
    }

    private void combinationSum(int[] nums, int target, int i, List<List<Integer>> res, List<Integer> combo) {
        if (target <= 0 || i >= nums.length) {
            if (target == 0) {
                res.add(new ArrayList(combo));
            }
            return;
        }
  
        int num = nums[i];
        combo.add(num);
        combinationSum(nums, target - num, i, res, combo);
        combo.remove(combo.size() - 1);
        combinationSum(nums, target, i + 1, res, combo);

    }



}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        longest = 0
        for num in nums:
            i = num
            length = 0
            while i in num_set:
                i += 1
                length += 1
            if length > longest:
                longest = length
        return longest


        